### PR TITLE
docs: Improve developer API + integration tests docs

### LINKED ISSUE #7660: (WIP) Improve  keptn documentation for developers
- [x] Add instructions on setting up gitea and mockserver for local integration tests 
- [x] List needed environmental variables: KEPTN_API_ENDPOINT KEPTN_API_TOKEN
- [x] Swagger Documentation uses the correct models (e.g. `/uniform/registration/{integrationID}/subscription` has no model for the 200 case although it returns the subscription id)
- [x] Improve documentation for resource-service: benefits & how to enable

### PR DESCRIPTION
**Pull request by odubajDT**
## This PR

- updates docs for API endpoints
- updates integration tests devel docs

### Related Issues

Part of #7660 

### Integration tests
https://github.com/keptn/keptn/actions/runs/2338804634

### CODE REVIEW

File: shipyard-controller/docs/swagger.yaml [hunk 1]
```diff
@@ -154,12 +158,24 @@ definitions:
...
         description: Creation date of the project
         type: string
+      gitProxyScheme:
+        description: git proxy scheme
+        type: string
+      gitProxyUrl:
+        description: git proxy URL
+        type: string
+      gitProxyUser:
+        description: git proxy user
```
Comment (thisthat) [inline shipyard-controller/docs/swagger.yaml:168]:
```suggestion
        description: Git proxy user
```

Comment (thisthat) [reply shipyard-controller/docs/swagger.yaml:168]: Also everywhere in the file. Please do a `s/git/Git`

File: shipyard-controller/docs/swagger.yaml [hunk 2]
```diff
@@ -154,12 +158,24 @@ definitions:
...
+        type: string
       gitRemoteURI:
         description: Git remote URI
         type: string
       gitUser:
         description: Git User
         type: string
+      insecureSkipTLS:
```
Comment (thisthat) [inline shipyard-controller/docs/swagger.yaml:176]: Why do we need to repeat it here? It is also at line 73

File: secret-service/pkg/handler/secrethandler.go [hunk 3]
```diff
@@ -35,8 +35,9 @@ type SecretHandler struct {
 // @Accept json
 // @Produce json
 // @Param secret body model.Secret true "The new secret to be created"
-// @Success 200 {object} model.Secret
+// @Success 201 {object} model.Secret
 // @Failure 400 {object} model.Error
+// @Failure 409 {object} model.Error
```
[the remark this answers is not in the document]
Comment (thisthat) [reply secret-service/pkg/handler/secrethandler.go:40]: Example: https://github.com/keptn/keptn/pull/7771/files#diff-0f941e2180f992c655e1cd2ab0e70303f71b4b96804b41899785a19bb681520cR65

**Review by thisthat — CHANGES_REQUESTED** [review]

File: shipyard-controller/docs/swagger.yaml (hunk 2 above)
Comment (odubajDT) [reply shipyard-controller/docs/swagger.yaml:176]: it was auto-generated by swagger, but we can remove it

Comment (odubajDT) [reply shipyard-controller/docs/swagger.yaml:176]: it's part of different models

File: shipyard-controller/docs/swagger.json [hunk 4]
```diff
@@ -1749,6 +1763,10 @@
                     "description": "git user",
                     "type": "string"
                 },
+                "insecureSkipTLS": {
+                    "description": "insecure skip tls\nomitempty property is missing due to fallback of this\nparameter to \"undefined\" when marshalling/unmarshalling data\nwhen \"false\" value is present",
```
Comment (thisthat) [inline shipyard-controller/docs/swagger.json:1767]: Could you please adapt also in this file the new message for `insecureSkipTLS`?

**Review by thisthat — APPROVED** [review]

File: docs/developer/install_gitea.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+ssh-keygen -t rsa -C "gitea-http" -f "rsa_gitea" -P "myGiteaPassPhrase"
+GITEA_PRIVATE_KEY=$(cat rsa_gitea)
+GITEA_PUBLIC_KEY=$(cat rsa_gitea.pub)
+GITEA_PRIVATE_KEY_PASSPHRASE=myGiteaPassPhrase
+
+sleep 30
+
+kubectl port-forward -n keptn svc/gitea-http 3000:3000 &
+kubectl port-forward -n keptn svc/gitea-ssh 3001:22 &
```
[the remark this answers is not in the document]
Comment (odubajDT) [reply docs/developer/install_gitea.sh:24]: I would say no, if we extract it to a separate step, we need to adapt the whole script, as in the next curl commands, we are using the services via port-forward https://github.com/keptn/keptn/pull/7771/files#diff-9fd7ebf47fd7154ae9c95d223cd4fbfcf12edd6a9097876fc0e6dab61aa1ec9fR28

File: shipyard-controller/docs/swagger.json (hunk 4 above)
Comment (odubajDT) [reply shipyard-controller/docs/swagger.json:1767]: adapted, thanks!

**Review by thisthat — APPROVED** [review]

**Pull request merged**